### PR TITLE
[android] Upgrade Gradle

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

To support app bundles, to get rid of the Android Studio “upgrade Gradle” dialog. Closes https://github.com/expo/expo/pull/2446.

# How

- upgraded Gradle plugin to 3.2.1
- upgraded Gradle wrapper to 4.10.2
- upgraded CreditCardEntry to 1.5.0, ZXing to 3.3.3
- removed `buildToolsVersion` from every `build.gradle` file as `buildToolsVersion` is not governed by Gradle plugin
- removed deprecated dependencies syntax (`compile`, `provided`) wherever possible
- added missing dependencies to `expoview-abi31_0_0`
- removed `minSdk` flavor dimension

# Test Plan

Expo Client compiles in both development and release mode. NCL and `Learn Swedish` apps work.

### CODE REVIEW

File: android/versioned-abis/expoview-abi25_0_0/build.gradle [hunk 1]
```diff
@@ -104,7 +103,7 @@ dependencies {
   api 'com.google.android.gms:play-services-vision:15.0.2'
   api 'com.google.android.gms:play-services-wallet:15.0.1' //may need 10.+
   api 'com.stripe:stripe-android:2.1.+'
-  api 'com.github.tipsi:CreditCardEntry:1.4.8.10'
+  api 'com.github.tipsi:CreditCardEntry:1.5.0'
```
Comment (bbarthec) [inline android/versioned-abis/expoview-abi25_0_0/build.gradle]: Is it reachable? I'm asking, cause I had to build it manually and provide via `.aar` file.

Comment (sjchmiela) [reply android/versioned-abis/expoview-abi25_0_0/build.gradle]: Yes, it's fetched via `jitpack.io`. BTW it was released 17 days ago, so you may have had to add AAR manually.

**Review by bbarthec — APPROVED** [review]

**Comment by sjchmiela** [comment]
Adding @nicknovitski as reviewer, if the PR passes “SDK” review we'll have to look into why CI failed with
```
Caused by: java.io.IOException: Cannot run program "/root/.gradle/caches/transforms-1/files-1.1/aapt2-3.2.1-4818971-linux.jar/1ca9a5f5e9da9e890fca1a2c181b24fe/aapt2-3.2.1-4818971-linux/aapt2": error=2, No such file or directory
```
error. 😕

**Comment by nicknovitski** [comment]
I have my suspicions; I've seen similar errors when the interpreter of packaged binaries is not present.  I'm looking into the problem.

**Review by esamelson — APPROVED** [review]
lgtm, thanks for taking care of this, sorry for letting it slip by last time around 😭

**Comment by nicknovitski** [comment]
Yeah, the problem is not reproducible on a mac; I can run `nix-shell --command ./build.sh` successfully.  I'm going to mess around with it on CI.

**Comment by nicknovitski** [comment]
Yep:
```
file /root/.gradle/caches/transforms-1/files-1.1/aapt2-3.2.1-4818971-linux.jar/1ca9a5f5e9da9e890fca1a2c181b24fe/aapt2-3.2.1-4818971-linux/aapt2
/root/.gradle/caches/transforms-1/files-1.1/aapt2-3.2.1-4818971-linux.jar/1ca9a5f5e9da9e890fca1a2c181b24fe/aapt2-3.2.1-4818971-linux/aapt2: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.24, not stripped
```

Specifically `interpreter /lib64/ld-linux-x86-64.so.2`.  That file is the one not being found.

What I don't know is why gradle is caching and running that particular aapt2 binary when one is already installed in the environment.
```
# nix-shell --command "which aapt2"
/nix/store/snalcv4kjd1hn6cip6hcgdyigw1d14lp-android-sdk-26.1.1/bin/aapt2
```

**Comment by nicknovitski** [comment]
A super-lame change that fixes the error: Adding `android.enableAapt2=false` to `gradle.properties`. But why would it not use the aapt2 already installed?  What task is running it?

**Comment by nicknovitski** [comment]
A slightly less lame change which also fixes the problem, creating the lib64 directory with an additional CI step: `nix build nixpkgs.stdenv.cc.libc --no-link && ln -s $(nix path-info nixpkgs.stdenv.cc.libc)/lib64 /lib64`

**Comment by nicknovitski** [comment]
Google searches find people running into this problem when they run builds in alpine-based docker containers.  They solve it by adding the `/lib64` path.

This is annoying enough that it makes me want to tear out the nix approach entirely, but we should be able to unblock this PR for now by adding the ci step `run: nix build nixpkgs.stdenv.cc.libc --no-link && ln -s $(nix path-info nixpkgs.stdenv.cc.libc)/lib64 /lib64` to the android client job.  I'd recommend also including an explanatory comment.

**Review by nicknovitski — APPROVED** [review]
thanks for doing this

**Pull request merged**